packages/ui-patterns/AssistantChat/AssistantCommandsPopover.tsx

```diff
@@ -0,0 +1,204 @@
+'use client'
+
+// import { AiIcon } from 'ui/components/Command'
+import {
+  Command,
+  CommandEmpty,
+  CommandGroup,
+  CommandInput,
+  CommandItem,
+  CommandList,
+  CommandSeparator,
+} from './../../ui/src/components/shadcn/ui/command'
+import { Popover, PopoverAnchor, PopoverContent } from './../../ui/src/components/shadcn/ui/popover'
+import { cn } from './../../ui/src/lib/utils/cn'
+import { createRef, useEffect, useRef, useState } from 'react'
+
+const AssistantCommandsPopover = ({
+  children,
+  textAreaRef,
+  setValue,
+  value,
+  open,
+  setOpen,
+  suggestions,
+  usingIcon = false,
+}: {
+  children: React.ReactNode
+  textAreaRef: React.RefObject<HTMLTextAreaElement>
+  setValue: (value: string) => void
+  value: string
+  open: boolean
+  setOpen: (value: boolean) => void
+  suggestions?: string[]
+  usingIcon?: boolean
+}) => {
+  const [command, setCommand] = useState<string>('')
+
+  const ref = createRef<HTMLDivElement>()
+  const commandWidth = ref.current?.clientWidth
+
+  const targetInputRef = useRef<HTMLInputElement | null>(null)
+
+  useEffect(() => {
+    // Attach the event listener when the component mounts
+    if (textAreaRef.current) {
+      textAreaRef.current.addEventListener('keydown', handleKeyPress)
+    }
+
+    // Detach the event listener when the component unmounts
+    return () => {
+      if (textAreaRef.current) {
+        textAreaRef.current.removeEventListener('keydown', handleKeyPress)
+      }
+      // Detach the event listener when the component unmounts
+    }
+  }, []) // Empty dependency array means this effect runs once when the component mounts
+
+  const handleKeyPress = (event: KeyboardEvent) => {
+    // console.log('event.key', event.key)
+    if (event.key === '/' && !value) {
+      // Add your action here
+      setOpen(true)
+      if (textAreaRef) textAreaRef?.current?.focus()
+    } else if (event.key === 'Escape') {
+      // Add your action here
+      setOpen(false)
+    } else if (event.key === 'Backspace') {
+      // Add your action here
+      setOpen(false)
+    } else {
+      if (event.key === 'ArrowUp' || event.key === 'ArrowDown' || event.key === 'Enter') {
+        // Forward the event to the target input
+        if (targetInputRef.current) {
+          const keyboardEvent = new KeyboardEvent('keydown', {
+            bubbles: true,
+            cancelable: true,
+            composed: true,
+            key: event.key,
+            code: event.code,
+          })
+
+          targetInputRef.current.dispatchEvent(keyboardEvent)
+
+          // Schedule focus on the original input using requestAnimationFrame
+          requestAnimationFrame(() => {
+            if (textAreaRef.current) {
+              textAreaRef.current.focus()
+            }
+          })
+
+          // Prevent the default behavior for ArrowUp, ArrowDown, and Enter
+          event.preventDefault()
+        }
+      }
+    }
+  }
+
+  useEffect(() => {}, [command, ref])
```

Comment:
Is this needed?

```suggestion
```

Reply:
cant remember why this is here. will test. wrote this all a while ago

Reply:
removed it